MovieCave - MVVM/MovieCave/MovieCave/VIew/TVSeriesView/TVSeriesViewModel.swift

```diff
@@ -0,0 +1,188 @@
+        self.coordinator = coordinator
+        self.movieDBService = movieDBService
+        self.currentPage = currentPage
+        self.list = list
+        fetchSeries(withFilter: list, on: currentPage)
+        setUpData()
+    }
+    
+    //MARK: - Methods
+    func performSearch(with searchText: String) {
+        guard !searchText.isEmpty else {
+            seriesArray.removeAll()
+            currentOperationType = .fetchSeries
+            fetchSeries(withFilter: list, on: currentPage)
+            return
+        }
+        
+        seriesArray.removeAll()
+        currentSearchText = searchText
+        currentOperationType = .searchSeriesByTitle
+        searchTVSeries(with: searchText, on: currentPage)
+    }
+    
+    func filterSeries(_ filter: String) {
+        switch filter {
+        case Constants.popularTVSeriesFilterButton:
+            list = .popular
+            fetchSeries(withFilter: .popular, on: currentPage)
+        case Constants.airingTodayTVSeriesFilterButton:
+            list = .airingToday
+            fetchSeries(withFilter: .airingToday, on: currentPage)
+        case Constants.topRatedTVSeriesFilterButton:
+            list = .topRated
+            fetchSeries(withFilter: .topRated, on: currentPage)
+        case Constants.onTheAirTVSeriesFilterButton:
+            list = .onTheAir
+            fetchSeries(withFilter: .onTheAir, on: currentPage)
+        default:
+            break
+        }
+    }
+
+    func sendSeriesDetails(with seriesID: Int) {
+        coordinator?.loadSeriesDetailsView(with: seriesID)
+    }
+
+    //MARK: - Private
+    private func fetchSeries(withFilter: TVSeriesLists, on page: Int) {
+        movieDBService.tvSeriesArrayCreation(with: withFilter.tvSeriesList(), on: page, operationType: .fetchSeries) { [weak self] result in
+            guard let self else { return }
+            
+            switch result {
+            case .success(let series):
+                guard !series.isEmpty else {
+                    self.fetchSeries(withFilter: withFilter, on: page - 1)
+                    return }
+                
+                self.seriesArray.append(contentsOf: series)
+                self.dataSource.items = self.seriesArray
+                self.fetchingDataSuccession.send(true)
+            case .failure(let error):
+                self.popUpMessage.send(error.localizedDescription)
+            }
+        }
+    }
+    
+    private func searchTVSeries(with query: String, on page: Int) {
+        movieDBService.tvSeriesArrayCreation(with: query, on: page, operationType: .searchSeriesByTitle) { [weak self] result in
+            guard let self else { return }
+            
+            switch result {
+            case .success(let series):
+                guard !series.isEmpty else {
+                    self.searchTVSeries(with: query, on: page - 1)
```

Comment:
Why do you need the 'page - 1' here? It's a little bit confusing. Either add a comment explaining or create a property with a good enough name that will be self explanatory